src/main/java/com/jcabi/log/ObjectDecor.java

```diff
@@ -72,41 +69,18 @@ public void formatTo(final Formatter formatter, final int flags,
         if (this.object == null) {
             formatter.format("NULL");
         } else if (this.object.getClass().isArray()) {
-            formatter.format(Arrays.deepToString((Object[]) this.object));
+            formatter.format(
+                AccessController.doPrivileged(
+                    new ArrayFormatAction((Object[]) this.object)
+                )
+            );
         } else {
             final String output =
-                AccessController.doPrivileged(new ObjectContentsFormatAction());
+                AccessController.doPrivileged(
+                    new ObjectContentsFormatAction(this.object)
+                );
             formatter.format(output);
         }
     }
 
-    /**
-     * {@link PrivilegedAction} for obtaining object contents.
-     */
-    private final class ObjectContentsFormatAction
```

Comment:
why did you decide to extract this class in separate file? If there is no weighty reason of doing this we need to keep it inside, as it was before